docs/pipelines/index.md

```diff
@@ -134,3 +134,199 @@ else
+    participant P as Pipeline
+    participant R as Retry
+    participant T as Timeout
+    participant D as DecoratedUserCallback
+
+    C->>P: Calls ExecuteAsync
+    P->>R: Calls ExecuteCore
+    R->>T: Calls ExecuteCore
+    Note over R, D: Initial attempt
+    Note over T: Wait start
+    activate T
+    T->>D: Invokes
+    activate D
+    D-->>D: Performs <br/>long-running <br/>operation
+    Note over T: Wait end
+    deactivate T
+    T-->>T: Times out
+    T->>D: Propagates cancellation
+    deactivate D
+    T->>R: Throws <br/>TimeoutRejectedException
+    R-->>R: Sleeps
+
+    Note over R, D: First retry attempt
+    R->>T: Calls ExecuteCore
+    Note over T: Wait start
+    activate T
+    T->>D: Invokes
+    activate D
+    D-->>D: Performs <br/>long-running <br/>operation
+    D->>T: Returns result
+    deactivate D
+    deactivate T
+    Note over T: Wait end
+
+    T->>R: Returns result
+    R->>P: Returns result
+    P->>C: Returns result
+```
+
+### Variant two
+
+<!-- snippet: resilience-pipeline-diagram-variant-2 -->
+```cs
+ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
+    .AddTimeout(TimeSpan.FromSeconds(10)) // outer
+    .AddRetry(new()) // inner
+    .Build();
+```
+<!-- endSnippet -->
+
+Lets suppose that the 1st and the 2nd requests are failing. The 3rd requested is not awaited since the overarching timeout elapsed.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    actor C as Caller
+    participant P as Pipeline
+    participant T as Timeout
+    participant R as Retry
+    participant D as DecoratedUserCallback
+
+    C->>P: Calls ExecuteAsync
+    P->>T: Calls ExecuteCore
+    Note over T: Wait start
+    activate T
+
+    T->>R: Calls ExecuteCore
+    Note over R, D: Initial attempt
+    R->>D: Invokes
+    activate D
+    D-->>D: Performs <br/>operation
+    D->>R: Fails
+    deactivate D
+
+    R-->>R: Sleeps
+    Note over R, D: First retry attempt
+    R->>D: Invokes
+    activate D
+    D-->>D: Performs <br/>operation
+    D->>R: Fails
+    deactivate D
+
+    R-->>R: Sleeps
+    Note over R, D: Second retry attempt
+    R->>D: Invokes
+    activate D
+    D-->>D: Performs <br/>operation
+
+    deactivate T
+    Note over T: Wait end
+    T-->>T: Times out
+    T->>R: Propagates cancellation
+    R->>D: Propagates cancellation
+    deactivate D
+
+    T->>P: Throws <br/>TimeoutRejectedException
+    P->>C: Propagates exception
+```
+
+### Variant three
+
+<!-- snippet: resilience-pipeline-diagram-variant-3 -->
+```cs
+ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
+    .AddTimeout(TimeSpan.FromSeconds(10)) // outer most
+    .AddRetry(new() { ShouldHandle = new PredicateBuilder().Handle<TimeoutRejectedException>() })
+    .AddTimeout(TimeSpan.FromSeconds(1)) // inner most
+    .Build();
+```
+<!-- endSnippet -->
+
+Lets suppose that the 1st request fails; the 2nd requests takes too long; and the 3rd requested is not awaited since the overarching timeout elapsed.
```

Comment:
```suggestion
Let's suppose that the first request fails and the second requests takes too long. The third requested is not awaited since the overarching timeout elapsed.
```